shell/browser/api/atom_api_web_contents.cc

```diff
@@ -1778,10 +1778,14 @@ void WebContents::Print(gin_helper::Arguments* args) {
   options.Get("landscape", &landscape);
   settings.SetBoolKey(printing::kSettingLandscape, landscape);
 
-  // We set the default to empty string here and only update
-  // if at the Chromium level if it's non-empty
+  // We set the default to the system's default printer and only update
+  // if at the Chromium level if the user overrides.
+  auto print_backend = printing::PrintBackend::CreateInstance(
+      nullptr, g_browser_process->GetApplicationLocale());
+  std::string default_printer = print_backend->GetDefaultPrinterName();
```

Comment:
This is a network operation, can you make it async with `base::PostTaskAndReplyWithResult` , check `LocalPrinterHandlerDefault::GetDefaultPrinter` for example.